src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -117,22 +119,33 @@ internal static unsafe partial class Utf8Utility
                         // the alignment check consumes at most a single DWORD.)
 
                         byte* pInputBufferFinalPosAtWhichCanSafelyLoop = pFinalPosWhereCanReadDWordFromInputBuffer - 3 * sizeof(uint); // can safely read 4 DWORDs here
-                        uint mask;
+                        int trailingZeroCount;
 
                         do
                         {
-                            if (Sse2.IsSupported)
+                            // pInputBuffer is 32-bit aligned but not necessary 128-bit aligned, so we're
+                            // going to perform an unaligned load. We don't necessarily care about aligning
+                            // this because we pessimistically assume we'll encounter non-ASCII data at some
+                            // point in the not-too-distant future (otherwise we would've stayed entirely
+                            // within the all-ASCII vectorized code at the entry to this method).
+                            if (AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian)
                             {
-                                // pInputBuffer is 32-bit aligned but not necessary 128-bit aligned, so we're
-                                // going to perform an unaligned load. We don't necessarily care about aligning
-                                // this because we pessimistically assume we'll encounter non-ASCII data at some
-                                // point in the not-too-distant future (otherwise we would've stayed entirely
-                                // within the all-ASCII vectorized code at the entry to this method).
+                                ulong mask = GetNonAsciiBytes(AdvSimd.LoadVector128(pInputBuffer));
+                                if (mask != 0)
+                                {
+                                    trailingZeroCount = BitOperations.TrailingZeroCount(mask);
+                                    Debug.Assert(trailingZeroCount < 16);
```

Comment:
@kunalspathak the System.Text.Json tests seem to be running some of this code and they failed by hitting this assert.
https://helix.dot.net/api/2019-06-17/jobs/689cf0fc-b595-46f8-902a-9c74e43509fb/workitems/System.Text.Json.Tests/console

Reply:
That's because you have missed to divide it by 4. 
See this for reference: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AdvSimdHelper.cs#L130

Reply:
Ah, right, thanks, updating.